### DISCUSSION TITLE
Recommendation for open source models for function calling.

### DISCUSSION THREAD

**Discussion by nyo16** [brainlid/langchain#160] [Q&A]
Hi, I know that the library is more optimized towards OAI models but I was wondering if anyone have tried to use Ollama and llama 3 or nous-hermes2pro-llama3-8b for function calling. 
I tried with some prompt variations but not much luck.

**Comment by cpursley** [comment]
I've had good luck with firefunction 2: https://fireworks.ai/blog/firefunction-v2-launch-post

https://huggingface.co/fireworks-ai/llama-3-firefunction-v2

**Comment by brainlid** [comment]
Check out this PR: https://github.com/brainlid/langchain/pull/198

It is in v0.3.0
